### DISCUSSION TITLE
CORS error when making a GraphQL request to Supabase

### DISCUSSION THREAD

**Discussion by BeetrootShoulders** [supabase/supabase#12288] [Questions]
I'm trying to query my Supabase database via GraphQL using urql in a Sveltekit application.

```
import { createClient, gql, queryStore } from '@urql/svelte';

const client = createClient({
    url: 'https://my-endpoint.supabase.co',
    fetchOptions: () => {
        return {
            headers: {
                apikey:'my-api-key',
                authorization:'Bearer my-api-key',
                'Access-Control-Allow-Origin': '*',
                'Access-Control-Allow-Headers': 'authorization, x-client-info, apikey, content-type'
            }
        };
    }
});

export async function load() {
    const { data } = queryStore({
        client: client,
        query: gql`
            query {
                todosCollection {
                    edges {
                        node {
                            id
                            title
                            description
                        }
                    }
                }
            }
        `
    });

    console.log(data);
    return {
        todos: data ?? []
    };
}
```

However, this throws a CORS error in the browser:

`(Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 404.`

When I check the headers, the Access-Control-Allow-Origin header is there exactly as I set it, although it is all in lower-case, though I assume that wouldn't make a difference. What am I doing wrong?

If I use the `supabase-js` client to make a 'regular' request, it of course works fine, but the headers show the `Access-Control-Allow-Origin` header as a _response_ header, as opposed to a _request_ header as it appears with my `urql` request. Is that the issue?

**Comment by GaryAustin1** [comment]
Normally the browser generate any needed CORS headers and preflight to initiate a request if it needs it.  The server then responds with what is allowed.   If the request is coming server side then no CORS is used.

**Reply by BeetrootShoulders** [reply]
Ok. If I remove all the headers and just set up the client with a url, then I get a 401 error, Unauthorized. If I just include the apikey or authorization fields, I get the CORS error again.

If I switch my `+page.js` file to `+page.server.js` to force it to run on the server, then I get no CORS errors, but my console log of `data` shows undefined, which I suppose is progress of a kind, but not very helpful in the long run. There are no errors of any kind in the console or network tab.

I asked a more general question yesterday regarding GraphQL + Supabase in Sveltekit but there have been no replies. There is basically zero information online about it - seems to be a bit of black hole at the moment.
